ledger/src/blockstore_processor.rs

```diff
@@ -1568,11 +1563,45 @@ pub fn fill_blockstore_slot_with_ticks(
     last_entry_hash
 }
 
+/// Check to see if the transactions exceeded the accounts data size limits
+fn check_accounts_data_size<'a>(
+    bank: &Bank,
+    execution_results: impl IntoIterator<Item = &'a TransactionExecutionResult>,
+) -> Result<()> {
+    check_accounts_data_block_size(bank)?;
+    check_accounts_data_total_size(bank, execution_results)
+}
+
+/// Check to see if transactions exceeded the accounts data size limit per block
+fn check_accounts_data_block_size(bank: &Bank) -> Result<()> {
+    if !bank
+        .feature_set
+        .is_active(&feature_set::cap_accounts_data_size_per_block::id())
+    {
+        return Ok(());
+    }
+
+    debug_assert!(MAX_ACCOUNT_DATA_BLOCK_LEN <= i64::MAX as u64);
+    if bank.load_accounts_data_size_delta_on_chain() > MAX_ACCOUNT_DATA_BLOCK_LEN as i64 {
```

Comment:
Not familiar with what `bank.load_accounts_data_size_delta_on_chain()`, but the logic looks good. Will deep dive to that function late.

Reply:
The change in accounts data size from the bank processing transactions is tracked in `Bank::accounts_data_size_delta_on_chain`, an `AtomicI64`. So ReplyStage here will check to ensure the block hasn't allocated more accounts data this block that the max (`MAX_ACCOUNT_DATA_BLOCK_LEN`), and only due to transactions.

This purposely ignores any other changes to the accounts data size, like reclaimed accounts due to rent collection (which are tracked in `Bank::accounts_data_size_delta_off_chain`).